### ISSUE TITLE
Full Rewrite using custom components

### ISSUE THREAD

**Issue by Orasund** [Orasund/elm-ui-widgets#19]
After using custom components for the first time, i question the usefulness of this package.

Before actuall doing the rewrite it might be useful to see how custom components and elm-ui interact.

**Comment by Orasund** [comment]
After looking into the code over at [aforemny/material-components-web-elm](https://package.elm-lang.org/packages/aforemny/material-components-web-elm/latest/), here are my conclusions:

* Customization work using `Html.Attributes` this means that a lot of the Elm-Ui magic will probably not work.
    This is a fundamental property of custom components. Unless Elm-Ui includes them as a native Element, this is most likely not possible to do in Elm-Ui.
* Global state and stateful components can be done without the user's notice. In particular the Modal Component is way better than anything that elm-ui-widgets could ever create.

I feel like if someone wants to use material design, i'd be best to point them towards Elm-Css instead of Elm-Ui.

In conclusion, I don't believe a full rewrite would be necessary. But maybe continuing with Material design might be a mad decision.
*
